### PR TITLE
vpa-recommender: Group prometheus and external-metrics flags to dedicated var blocks

### PR DESCRIPTION
**Pull request by ialidzhikov**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Right now, it is clear where the prometheus and external-metrics flags start but it is not clear where they end.
For example, `--memory-saver` is not a prometheus history provider related flag but it appears so.
This PR uses the already existing pattern of grouping flags into dedicated var blocks (see the rest of the flags; there are other groups like aggregation configuration flags and post processor flags).

#### Which issue(s) this PR fixes:

N/A

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by raywainman** [comment]
/lgtm

Thanks for the cleanup

File: vertical-pod-autoscaler/pkg/recommender/main.go [hunk 1]
```diff
@@ -62,9 +62,12 @@ var (
 	prometheusAddress      = flag.String("prometheus-address", "http://prometheus.monitoring.svc", `Where to reach for Prometheus metrics`)
 	prometheusJobName      = flag.String("prometheus-cadvisor-job-name", "kubernetes-cadvisor", `Name of the prometheus job name which scrapes the cAdvisor metrics`)
```
Comment (adrianmoisey) [inline vertical-pod-autoscaler/pkg/recommender/main.go:63]: Would it make sense to move these two flags into the prometheus block also? May be put them at the start.

**Comment by adrianmoisey** [comment]
/lgtm

I'm a huge fan of changes like this, they seem super small, but make the project that much easier to work on. Over time these add up making for a nice neat easy to work on project. Thanks for doing it!

**Comment by voelzmo** [comment]
/approve

Thanks!

**Pull request merged**